scripts/Connector/Connector_Hirose/conn_hirose_df11_tht_top.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python3
+
+import sys
+import os
+#sys.path.append(os.path.join(sys.path[0],"..","..","kicad_mod")) # load kicad_mod path
+
+# export PYTHONPATH="${PYTHONPATH}<path to kicad-footprint-generator directory>"
+sys.path.append(os.path.join(sys.path[0], "..", "..", ".."))  # load parent path of KicadModTree
+from math import sqrt
+import argparse
+import yaml
+from helpers import *
+from KicadModTree import *
+
+sys.path.append(os.path.join(sys.path[0], "..", "..", "tools"))  # load parent path of tools
+from footprint_text_fields import addTextFields
+
+series = ""
+series_long = 'DF11 through hole'
+manufacturer = 'Hirose'
+orientation = 'V'
+number_of_rows = 2
+datasheet = 'https://www.hirose.com/product/document?clcode=&productname=&series=DF11&documenttype=Catalog&lang=en&documentid=D31688_en'
+
+#pins_per_row per row
+pins_per_row_range = range(2,17)
+
+#Molex part number
+#n = number of circuits per row
+part_code = "DF11-{n}DP-2DSA"
+
+pitch = 2
+drill = 0.8
```

Comment:
For 0.5mm square pins, that are ~0.7mm diagonal, IPC 2221 would say use 0.2mm oversize for the drill or 0.9mm. But if the datasheet tells us something different we go with that. In this case, the datasheet shows 0.8mm -0/+0.1mm which is a nominal 0.85mm. So please use as a the drill size.